src/BinSkim.Driver/RoslynAnalyzer/RoslynCompilationStartAnalysisContext.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.IL
+{
+    /// <summary>
+    /// Basic analysis context provided to Roslyn analyzers. We use a singleton instance of this class to 
+    /// capture all symbol action registration for analyzers. These actions will subsequently be invoked
+    /// as we visit the IL of all analysis targets.
+    /// </summary>
+    internal sealed class RoslynCompilationStartAnalysisContext : CompilationStartAnalysisContext
+    {
+        public ActionMap<SymbolAnalysisContext, SymbolKind> SymbolActions { get; set; }
+
+        public Action<CompilationAnalysisContext> CompilationEndActions { get; set; }
```

Comment:
Ditto. Why public setter? I would expect these to be ReadOnly with RegisterXxx the only way to add new actions.